### PR TITLE
chart for vent

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: greg <greg.hill@monax.io>

### CODE REVIEW

File: src/vent/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command:
+            - "vent"
+          args:
+            - "--db-adapter"
+            - "postgres"
```
Comment (compleatang) [inline src/vent/templates/deployment.yaml]: expose this as a variable in values. or something like `{{ if .Values.postgresql.enabled }}postgres{{ else }}sqlite{{end}}`

File: src/vent/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            - "--db-url"
+            - "postgres://{{ .Values.postgresql.postgresUser }}:{{ .Values.postgresql.postgresPassword }}@{{ .Values.postgresql.postgresRelease }}\
```
Comment (compleatang) [inline src/vent/templates/deployment.yaml]: let's take into account the sqlite usage of the chart here as well.

File: src/vent/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+              :{{ .Values.postgresql.postgresPort }}/{{ .Values.postgresql.postgresDatabase }}"
```
Comment (compleatang) [inline src/vent/templates/deployment.yaml]: does splitting lines like this actually work?

File: src/vent/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            - "--db-schema"
+            - "{{ .Values.postgresql.postgresSchema }}"
```
Comment (compleatang) [inline src/vent/templates/deployment.yaml]: these postgres values are a bit redundant let's use, e.g, `{{ .Values.postgresql.schema}}`

File: src/vent/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            - "--grpc-addr"
+            - "{{ .Values.burrow.node }}:{{ .Values.burrow.port }}"
```
Comment (compleatang) [inline src/vent/templates/deployment.yaml]: `s/node/host/g`

File: src/vent/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            - "--log-level"
+            - "debug"
```
Comment (compleatang) [inline src/vent/templates/deployment.yaml]: expose this as a variable

File: src/vent/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            - "--spec-file"
+            - "/config/EventSpec.json"
```
Comment (compleatang) [inline src/vent/templates/deployment.yaml]: expose this as a variable and plan for the addition of the `--spec-dir` flag.

File: src/vent/templates/configmap.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+metadata:
+  name: {{ template "vent.fullname" . }}
+  labels:
+    app: {{ template "vent.name" . }}
+    chart: {{ template "vent.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  EventSpec.json: |
+    [
```
Comment (compleatang) [inline src/vent/templates/configmap.yaml]: let's use a `files` modality here. see, e.g., grafana chart or prometheus chart and how they deal with multiple files that get mounted as config maps.

File: src/vent/values.yaml [hunk 9]
```diff
@@ -0,0 +1,21 @@
+replicaCount: 1
+
+image:
+  repository: quay.io/monax/vent
+  tag: develop
+  pullPolicy: IfNotPresent
+
+postgresql:
+  enabled: true
+  postgresRelease: postgres
```
Comment (compleatang) [inline src/vent/values.yaml]: `host` ...? release sounds weird in this context.

File: src/vent/templates/deployment.yaml (hunk 3 above)
Comment (gregdhill) [reply src/vent/templates/deployment.yaml]: http://yaml.org/spec/1.2/spec.html#style/flow/double-quoted

File: src/blackstone/templates/deployment-vent.yaml [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+kind: Deployment
+metadata:
+  name: {{ template "blackstone.fullname" . }}-vent
+  labels:
+    app: {{ template "blackstone.name" . }}-vent
+    chart: {{ template "blackstone.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
```
Comment (compleatang) [inline src/blackstone/templates/deployment-vent.yaml]: `{{ .Values.vent.replicaCount }}`

File: src/blackstone/templates/deployment-vent.yaml [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+          command:
+            - "vent"
+          args:
+{{ if eq .Values.vent.backend.type "postgres" }}
+            - "--db-adapter"
+            - "postgres"
+            - "--db-url"
+            - "postgres://{{ .Values.vent.backend.postgresUser }}:{{ .Values.vent.backend.postgresPass }}@{{ .Values.vent.backend.postgresHost }}\
+              :{{ .Values.vent.backend.postgresPort }}/{{ .Values.vent.backend.database }}"
+{{ else if eq .Values.vent.backend.type "sqlite" }}
```
Comment (compleatang) [inline src/blackstone/templates/deployment-vent.yaml]: remove sqlite since this is not a standalone chart.

**Pull request merged**